File: Task 1/task1.js
function optimizeBookings(bookings) {
    let ans = [];

    if (bookings.length === 0) {
        return ans;
    }

    
    bookings.sort((a, b) => a[0] - b[0]);


    let tempint = bookings[0];

    for (let i = 1; i < bookings.length; i++) {
        let current = bookings[i];
        
        if (tempint[1] >= current[0]) {
            tempint[1] = Math.max(tempint[1], current[1]);
        } else {
            
            ans.push(tempint);
            tempint = current; 
        }
    }

    ans.push(tempint);

    return ans;
}

const n = prompt("Enter the number of bookings: ");
let bookings = [];

for (let i = 0; i < n; i++) {
    let start = prompt(`Enter the start time for booking ${i + 1}: `);
    let end = prompt(`Enter the end time for booking ${i + 1}: `);
    bookings.push([parseInt(start), parseInt(end)]);
}


let ans = optimizeBookings(bookings);

console.log("Optimized bookings: ");
ans.forEach(booking => {
    console.log(`[${booking[0]}, ${booking[1]}]`);
});

File: Task 1/task1.cpp
#include<bits/stdc++.h>
using namespace std;


vector<vector<int>> optimizeBookings(vector<vector<int>>& bookings) {
    vector<vector<int>> ans;

    if (bookings.empty()) {
        return ans;
    }

    
    sort(bookings.begin(), bookings.end());

    
    vector<int> tempint = bookings[0];

    for (const auto& it : bookings) {
        
        if (tempint[1] >= it[0]) {
            tempint[1] = max(tempint[1], it[1]);
        } else {
            
            ans.push_back(tempint);
            tempint = it; 
        }
    }
    
    ans.push_back(tempint);

    return ans;
}

int main() {
    int n;
    cout << "Enter the number of bookings: ";
    cin >> n;

    vector<vector<int>> bookings(n, vector<int>(2)); 
    cout << "Enter the bookings (start end): " << endl;

    for (int i = 0; i < n; i++) {
        cin >> bookings[i][0] >> bookings[i][1];
    }

    
    vector<vector<int>> ans = optimizeBookings(bookings);

    cout << "optimizeBookingsd bookings: " << endl;
    for (const auto& i : ans) {
        cout << "[" << i[0] << ", " << i[1] << "]" << endl;
    }

    return 0;
}
